src/components/academies/tests/AcademyDetailPage.test.jsx

```diff
@@ -130,4 +119,39 @@ describe('<AcademyDetailPage />', () => {
     renderWithRouter(<AcademyDetailPageWrapper />);
     expect(screen.getByTestId('not-found-page')).toBeInTheDocument();
   });
+  it.each(
+    generateTestPermutations({
+      shouldUseSecuredAlgoliaApiKey: [true, false],
+      searchClient: [null, { search: jest.fn().mockResolvedValue(ALOGLIA_MOCK_DATA) }],
```

Comment:
nit/consideration: would it make sense to create a variable for the searchClient mock to be a bit more DRY here, but also throughout these tests (e.g., in the `beforeEach`)?

```tsx
const mockSearchFn = jest.fn().mockResolvedValue(ALOGLIA_MOCK_DATA);
const mockSearchClient = { search: mockSearchFn };
```

Reply:
That said, worth sanity checking that the searchClient does indeed have a `search` function; I believe the `searchIndex` would have this `search` function, but the `searchClient` having properties like `initIndex`.